Lib/idlelib/autocomplete.py

```diff
@@ -183,10 +183,14 @@ def control_char_escape(match):
         def high_byte_sub(match):
             """escaping for bytes values above 127"""
             return f'\\x{ord(match.group()):x}'
+        non_bmp_re = re.compile(r'[\U00010000-\U0010ffff]')
```

Comment:
As suggested by my email, this and the function or something like them, and probably the code that calls it, is needed many other places.  But I don't know yet where it should go.  Thanks for the start.